flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/jsonplan/ConfigureOperatorLevelStateTtlJsonITCase.java

```diff
@@ -0,0 +1,205 @@
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.JsonNode;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for configuring operator-level state TTL via {@link
+ * org.apache.flink.table.api.CompiledPlan}.
+ */
+public class ConfigureOperatorLevelStateTtlJsonITCase extends JsonPlanTestBase {
+
+    @Test
+    public void testDifferentStateTtlForDifferentOneInputOperator() throws Exception {
+        String dataId =
+                TestValuesTableFactory.registerRowData(
+                        Arrays.asList(
+                                GenericRowData.of(1, StringData.fromString("Tom"), 1, 199.9d),
+                                GenericRowData.of(2, StringData.fromString("Jerry"), 2, 99.9d),
+                                GenericRowData.of(1, StringData.fromString("Tom"), 1, 199.9d),
+                                GenericRowData.of(3, StringData.fromString("Tom"), 1, 29.9d),
+                                GenericRowData.of(4, StringData.fromString("Olivia"), 1, 100d),
+                                GenericRowData.of(4, StringData.fromString("Olivia"), 1, 100d),
+                                GenericRowData.of(2, StringData.fromString("Jerry"), 2, 99.9d),
+                                GenericRowData.of(5, StringData.fromString("Michael"), 3, 599.9d),
+                                GenericRowData.of(6, StringData.fromString("Olivia"), 3, 1000d)));
+        createTestSourceTable(
+                "Orders",
+                new String[] {
+                    "`order_id` INT", "`buyer` STRING", "`quantity` INT", "`amount` DOUBLE"
+                },
+                null,
+                getProperties(dataId, 1, "2s"));
+
+        createTestNonInsertOnlyValuesSinkTable(
+                "OrdersStats",
+                "`buyer` STRING",
+                "`ord_cnt` BIGINT",
+                "`quantity_cnt` BIGINT",
+                "`total_amount` DOUBLE");
+        compileSqlAndExecutePlan(
+                        "INSERT INTO OrdersStats \n"
+                                + "SELECT buyer, COUNT(1) AS ord_cnt, SUM(quantity) AS quantity_cnt, SUM(amount) AS total_amount FROM (\n"
+                                + "SELECT *, ROW_NUMBER() OVER(PARTITION BY order_id, buyer, quantity, amount ORDER BY proctime() ASC) AS rk FROM Orders) tmp\n"
+                                + "WHERE rk = 1\n"
+                                + "GROUP BY buyer",
+                        json -> {
+                            try {
+                                JsonNode target = JsonTestUtils.readFromString(json);
+                                JsonTestUtils.setExecNodeStateMetadata(
+                                        target, "stream-exec-deduplicate", 0, 6000L);
+                                JsonTestUtils.setExecNodeStateMetadata(
```

Comment:
I'm wondering whether it'll be better to change it `8000L`.
The reason is that if we remove this code line, the test can still passs so that we can't make sure set state for `stream-exec-group-aggregate` also make difference.

Reply:
Commenting out this line means that the TTL of the deduplication node will take the default value of 0, which means that the status will never expire. However, the actual effect is the same as setting a TTL of 6 or 8 seconds.

I think this case demonstrates the significance of supporting operator-level TTL configuration. That is to say, as long as we ensure that the data calculation results remain unchanged, we do not need to unnecessarily extend the state TTL for the sake of a long-period state of a certain operator.

Let me add more description to the case.